### PR TITLE
feat(Eraser): v2 #2

### PR DESCRIPTION
**Pull request by ShaMan123**
### Summary
This PR along with #7470 consist of Eraser's v2.
This PR **deprecates** eraser's bad use of `clipPath`.
It was bad practice to do it as I did (at the time I was overwhelmed by fabric's internals), so know it's fixed and for the best!
This fix removes all awkwardness that was caused by riding on clip path (handling clip path and eraser was brittle).
This makes eraser much more robust, more legitimate and less a hack. **MATURE**. a proper part of fabric.
Previously introducing eraser to an existing project would break clip path in some cases and require a lot of touch up.
Now it doesn't.
Simply plug and play.

### Changes
1. clip path has returned to it's normal behavior.
2. Object has a dedicated prop `eraser` (instead of riding on clip path).
3. Eraser is now a subclass of group. This means that `fromJSON` will break between versions. Correct it by changing the `type` prop from `group` to `eraser` and remove the first (rect) object.
4. dropped `getEraser` `setClipPath` `getClipPath`
5. fixed eraser clipped by `absolutePositioned` clip path 
4. fixed `toClipPathSVG` -  support nested clip paths (*group still remains problematic #7586 )

### Next Up
Tests

### CODE REVIEW

File: src/shapes/object.class.js [hunk 1]
```diff
@@ -1958,6 +1958,15 @@
...
   fabric.Object.NUM_FRACTION_DIGITS = 2;
 
+  /**
+   * Defines which properties should be enlivened from the object passed to {@link fabric.Object._fromObject}
+   * @static
+   * @memberOf fabric.Object
+   * @constant
+   * @type string[] 
+   */
+  fabric.Object.ENLIVEN_PROPS = ['clipPath'];
```
Comment (asturur) [inline src/shapes/object.class.js:1968]: this is missing in the first PR.

File: src/mixins/object.svg_export.js [hunk 2]
```diff
@@ -180,15 +180,26 @@
...
+      var commonPieces = [
+        this.getSvgTransform(true, additionalTransform),
+        this.getSvgCommons(),
+      ].join(''),
+        // insert commons in the markup, style and svgCommons
+        index = objectMarkup.indexOf('COMMON_PARTS');
       objectMarkup[index] = commonPieces;
-      return reviver ? reviver(objectMarkup.join('')) : objectMarkup.join('');
+      markup.push(objectMarkup.join(''));
+      return reviver ? reviver(markup.join('')) : markup.join('');
```
Comment (asturur) [inline src/mixins/object.svg_export.js:202]: ok this is a tentative to make the clipPath svg export recursive, right?
This is unrelated to brush, is a general improvement/fix and deserve its own pr with test ( testing svg export is very easy luckly )
Sorry for the extra work but i prefer we split it.

File: src/shapes/object.class.js (hunk 1 above)
Comment (asturur) [reply src/shapes/object.class.js:1968]: Could FILL, STROKE be a part of the enlivable props and handle gradient and patterns?
maybe a tought for a later pr.

As today we have this discrepancy where you can send in the object options a non enlived gradient and pattern and the object will iniitialize them. While this does not work for clipPath.
I wonder if we should make the same for everything

Comment (ShaMan123) [reply src/shapes/object.class.js:1968]: Sounds easy to adjust

File: src/mixins/object.svg_export.js (hunk 2 above)
Comment (ShaMan123) [reply src/mixins/object.svg_export.js:202]: Alright

File: src/shapes/object.class.js (hunk 1 above)
Comment (ShaMan123) [reply src/shapes/object.class.js:1968]: And a good idea

Comment (ShaMan123) [reply src/shapes/object.class.js:1968]: > this is missing in the first PR.

I see it on both PRs

**Comment by ShaMan123** [comment]
@asturur do you mind I shut this PR down and work only on #7470 ?
It's easier for me beause some changes are here and some there and then I need to resolve conflict and so on.

**Comment by ShaMan123** [comment]
merged into #7470

**Comment by asturur** [comment]
ok i was about to ask
